[MODULAR] IRCD removal, Improved RCD added, also featuring, the AFAD

RatFromTheJungle opened the pull request:
## About The Pull Request
Since the implementation of guncargo, two things became very-clear, one, that departmental orders are too broad, and generally, can supply an entire department without issue, even for 'trivial' items, and that cargo has a surprising lack of stuff to buy, this is step one-in-a-few, to introduce stuff people have mentioned wanting in the past, while removing items that over perform every other option (the IRCD)

AFAD: A medibeam, that's very bad at it's job, but saves people having to use brute-pills or sutures for 5 brute, or because they grabbed a light-bulb.
![image](https://user-images.githubusercontent.com/62520989/181625432-c4b81d41-8c4f-46dd-8c92-b2139c35d6a1.png)


IRCD: Explained in the desc of the item, really, just an RCD with more storage, and a slower speed.

they're gonna start off pretty-pricey to see where they settle balance wise, before adjusting the price.

Might decide to mark this as a draft if I get a sudden idea for something better.

## How This Contributes To The Skyrat Roleplay Experience
Variety, no more IRCD's being a flat an objective improvement over RCD's without a single downside.

Also, removed those jukebox crates that were commented out, despite the fact no one is ever gonna re-add those.

## Changelog



:cl:
add: The AFAD to the cargo goodies section, a very-low power medibeam meant primarily for low-damage situations (15 or less total damage)
balance: Removes the IRCD from the cargo packs, and replaces it with an Improved-RCD goodie.

/:cl:

Zonespace27 commented on modular_skyrat/modules/cargo/code/items/AFAD.dm:
```diff
@@ -0,0 +1,151 @@
+// Copypasted just to avoid having to make non-modular edits to the medbeam.
```
bad

RatFromTheJungle replied:
https://www.masterclass.com/articles/how-to-write-descriptive-sentences

softcerv commented:
> You currently have a negative Fix/Feature pull request delta of -68. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

so close to greatness

NotRanged commented:
Issue: there are few meaningful things to buy from cargo

Solution: removed another one of them

lol

RatFromTheJungle commented:
> Issue: there are few meaningful things to buy from cargo
> 
> Solution: removed another one of them

*and added two separate ones, which are a much better middle-ground between mid-round items, and roundstart stuff, while removing something people usually mindlessly tick every 15m for an essentially, free, effortless powerup they were really never intended to have by how much it bricks any semblance of trying to make something secure; and made engie-antag even stronger then it normally is by essence of the role.


Finished the sentence for you :)

OrionTheFox commented on modular_skyrat/modules/cargo/code/goodies.dm:
```diff
@@ -23,6 +23,25 @@
+
+/datum/supply_pack/goody/improvedrcd
+	name = "Improved RCD"
+	desc = "An upgraded RCD featuring superior material storage, and a complementary frames, and circuits upgrade to boot!"
```
```suggestion
	desc = "An upgraded RCD featuring superior material storage, and complementary frames and circuits upgrades to boot!"
```

OrionTheFox commented on modular_skyrat/modules/cargo/code/goodies.dm:
```diff
@@ -23,6 +23,25 @@
+	
+/datum/supply_pack/goody/improvedrcdbonus
+	name = "Improved RCD  Value Pack"
+	desc = "A pack of 3 upgraded RCDs featuring superior material storage, and a complementary frames, and circuits upgrade to boot!"
```
```suggestion
	desc = "A pack of 3 upgraded RCDs featuring superior material storage, and complementary frames and circuits upgrades to boot!"
```

OrionTheFox commented on modular_skyrat/modules/cargo/code/goodies.dm:
```diff
@@ -111,3 +130,13 @@
+
+/datum/supply_pack/goody/afad
+	name = "Automated First Aid Device"
+	desc = "Someone hurt, and it's to much of a bother to manually handle their burns, or cuts? Look no further then the AFAD, a state of the art first-aid device!"
```
```suggestion
	desc = "Someone mildly hurt, and it's to much of a bother to manually handle their burns or cuts? Look no further then the AFAD, a state of the art pain-relief device!"
```
First-Aid implies its used for, well, first aid - treating life-threatening wounds. Considering its nerfed to only relieve minor damage, I dont think First-Aid applies.

GoldenAlpharex commented on modular_skyrat/modules/cargo/code/goodies.dm:
```diff
@@ -23,6 +23,25 @@
+	name = "Improved RCD  Value Pack"
+	desc = "A pack of 3 upgraded RCDs featuring superior material storage, and a complementary frames, and circuits upgrade to boot!"
+	cost = PAYCHECK_CREW * 100
+	contains = list(/obj/item/construction/rcd/combat/improved,
+					/obj/item/construction/rcd/combat/improved,
+					/obj/item/construction/rcd/combat/improved)
```
```suggestion
	contains = list(
		/obj/item/construction/rcd/combat/improved,
		/obj/item/construction/rcd/combat/improved,
		/obj/item/construction/rcd/combat/improved,
	)
```

OrionTheFox commented on modular_skyrat/modules/cargo/code/items/AFAD.dm:
```diff
@@ -0,0 +1,150 @@
+/obj/item/gun/afad
```
Do we need to recode this in its entirety? Can it not be a subtype of the medbeam?

RatFromTheJungle replied:
redone

GoldenAlpharex commented on modular_skyrat/modules/cargo/code/items/AFAD.dm:
```diff
@@ -0,0 +1,150 @@
+/obj/item/gun/afad
```
Is this just a copy-paste of the medigun code, but with a different typepath, just because you want to have a different healing behavior? Because if so, that's definitely not the way to go.

RatFromTheJungle replied:
redone

GoldenAlpharex commented on modular_skyrat/modules/cargo/code/items/improvedRCD.dm:
```diff
@@ -0,0 +1,11 @@
+// Slightly improved version of the normal RCD, mostly as an engineering 'I got hella bread' purchase
+/obj/item/construction/rcd/combat/improved
```
Why is it a subtype of the combat RCD, exactly?

RatFromTheJungle replied:
![image](https://user-images.githubusercontent.com/62520989/180626848-02cd64bb-d756-40a5-b15b-fc2d0c5e89ca.png)

real footage of my brain making this at 5am

tf-4 commented on modular_skyrat/modules/cargo/code/goodies.dm:
```diff
@@ -23,6 +23,27 @@
+
+/datum/supply_pack/goody/improvedrcd
+	name = "Improved RCD"
+	desc = "An upgraded RCD featuring superior material storage, and complementary frames and circuit upgrades to boot!"
```
```suggestion
	desc = "An upgraded RCD featuring superior material storage. Comes with complimentary frames and circuitry upgrades to boot!"
```

tf-4 commented on modular_skyrat/modules/cargo/code/goodies.dm:
```diff
@@ -23,6 +23,27 @@
+	
+/datum/supply_pack/goody/improvedrcdbonus
+	name = "Improved RCD  Value Pack"
+	desc = "A pack of 3 upgraded RCDs featuring superior material storage, and complementary frames and circuits upgrades to boot!"
```
```suggestion
	desc = "A pack of 3 upgraded RCDs featuring superior material storage. Each comes with complimentary frames and circuitry upgrades to boot!"
```

tf-4 commented on modular_skyrat/modules/cargo/code/goodies.dm:
```diff
@@ -111,3 +132,13 @@
+
+/datum/supply_pack/goody/afad
+	name = "Automated First Aid Device"
+	desc = "Someone mildly hurt, and it's to much of a bother to manually handle their burns or cuts? Look no further then the AFAD, a state of the art pain-relief device!"
```
```suggestion
	desc = "Someone mildly hurt and it's too much of a bother to manually handle their burns or cuts? Look no further than the AFAD, a state-of-the-art pain-relief device!"
```

tf-4 commented on modular_skyrat/modules/cargo/code/items/AFAD.dm:
```diff
@@ -0,0 +1,18 @@
+/obj/item/gun/medbeam/afad
+	name = "Automated First Aid Device"
+	desc = "Usually supplied in medkits, the AFAD is a revolutionary device, or more accurately, a knock-off of the legendary medibeam meant for fixing scrapes and bruises, a label reminds you not to cross the beams."
```
```suggestion
	desc = "Usually supplied in medkits, the AFAD is a revolutionary device meant for fixing scrapes and bruises, and totally not a knockoff of the legendary medibeam. A label on the underside reminds you not to cross the beams."
```

GoldenAlpharex commented on modular_skyrat/modules/cargo/code/items/AFAD.dm:
```diff
@@ -0,0 +1,18 @@
+	name = "Automated First Aid Device"
+	desc = "Usually supplied in medkits, the AFAD is a revolutionary device meant for fixing scrapes and bruises, and totally not a knockoff of the legendary medibeam. A label on the underside reminds you not to cross the beams."
+	icon = 'icons/obj/chronos.dmi'
+	icon_state = "chronogun"
+	inhand_icon_state = "chronogun"
```
Could we have pictures of what it looks like in-game?

RatFromTheJungle replied:
![image](https://user-images.githubusercontent.com/62520989/181624985-827a825e-f661-4cf6-b402-ebc0e95296c7.png)
it looks like a medbeam

GoldenAlpharex commented on modular_skyrat/modules/cargo/code/items/AFAD.dm:
```diff
@@ -0,0 +1,18 @@
+/obj/item/gun/medbeam/afad/on_beam_tick(mob/living/target)
+	if(target.health != target.maxHealth)
+		new /obj/effect/temp_visual/heal(get_turf(target), "#80F5FF")
+	target.adjustBruteLoss(-0.2)
+	target.adjustFireLoss(-0.2)
+	target.adjustToxLoss(-0.1)
+	target.adjustOxyLoss(-0.1)
```
Those values would be nice to have stored either as variables on the afad, or as defines.

RatFromTheJungle replied:
sadly, because it's a child of the medbeam, as you asked, no-can-do without it causing issues; or atleast, I was told when I asked about it. has to be done how it's done on the medbeam.

RatFromTheJungle replied:
attempted, bravely
